.github/workflows/publish_charm.yaml

```diff
@@ -152,21 +149,32 @@ jobs:
       - name: Install charmcraft
         run: |
           sudo snap install charmcraft --classic
+      - name: Get charm name
+        id: get-charm-name
+        run: echo "CHARM_NAME=$(yq -e '.name' metadata.yaml)" >> $GITHUB_ENV
       - name: Log in to the Container registry
         uses: docker/login-action@v2
         with:
           registry: ${{ env.REGISTRY }}
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Download images artifact
```

Comment:
I assume this is for testing purposes, am I right?

Reply:
The integration workflow will upload the image tags used in the workflow as an artifact, and in the publish workflow, the file contains images tags will be downloaded in order to use the same image as the integration test. Images are not in the artifact, only the names.

Reply:
I see. But the artifact won't be available on merge unless you download it using the API. Am I missing something here?

Reply:
The image is pushed to the GitHub container registry, so we only need the name to pull the image from Github container registry.

Reply:
Yes, I get you, but this artifact won't be available on merge. You'll need to follow an approach like https://github.com/canonical/operator-workflows/blob/main/.github/workflows/publish_charm.yaml#L188

Reply:
Gotcha, adding another step for downloading the images artifact with gh command, thanks!